chore: remove `SECONDS_PER_SLOT` and replace it with `network_spec().seconds_per_slot`

Dyslex7c opened the pull request:
### What was wrong?

Constant `SECONDS_PER_SLOT` being used.

### How was it fixed?

Removed `SECONDS_PER_SLOT` and used `network_spec().seconds_per_slot` instead.
Closes #637 

### To-Do

 <!-- Stay ahead of things, add list items here!  -->
- [ ] I have read [CONTRIBUTING.md](https://github.com/ReamLabs/ream/blob/master/CONTRIBUTING.md).
- [ ] This PR uses [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).

KolbyML commented:
instead of making a crate called ream-spec-utils

could you split 
``crates/common/consensus``

into
- ``crates/common/consensus/beacon``

^ the crates name will be `ream-consensus-beacon`

-  ``crates/common/consensus/misc``
^ the crates name will be `ream-consensus-misc``

<img width="194" height="356" alt="image" src="https://github.com/user-attachments/assets/1d61940b-1a1f-4d8a-8d03-0f68c296308f" />

^ you can move these structs to the new crate

Dyslex7c commented:
@KolbyML I'm getting this error
```
./target/debug/ream beacon_node
2025-07-14T10:25:04.392292Z  INFO ream: starting up beacon node...

thread 'tokio-runtime-worker' panicked at bin/ream/src/main.rs:92:49:
unable to init Ream Database: Redb(TableTypeMismatch { table: "beacon_block", key: TypeName { classification: UserDefined, name: "SSZEncoding<alloy_primitives::bits::fixed::FixedBytes<32>>" }, value: TypeName { classification: UserDefined, name: "SSZEncoding<ream_consensus::electra::beacon_block::SignedBeaconBlock>" } })
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I've updated `ream-consensus` to the new crates in every place and checked it using `grep`

KolbyML commented:
> @KolbyML I'm getting this error
> 
> ```
> ./target/debug/ream beacon_node
> 2025-07-14T10:25:04.392292Z  INFO ream: starting up beacon node...
> 
> thread 'tokio-runtime-worker' panicked at bin/ream/src/main.rs:92:49:
> unable to init Ream Database: Redb(TableTypeMismatch { table: "beacon_block", key: TypeName { classification: UserDefined, name: "SSZEncoding<alloy_primitives::bits::fixed::FixedBytes<32>>" }, value: TypeName { classification: UserDefined, name: "SSZEncoding<ream_consensus::electra::beacon_block::SignedBeaconBlock>" } })
> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> ```
> 
> I've updated `ream-consensus` to the new crates in every place and checked it using `grep`

do --purge-db while running

Dyslex7c commented:
@KolbyML I've made the changes, please do check the latest commit

Dyslex7c commented:
@KolbyML done, please check

KolbyML commented:
@Dyslex7c 
<img width="483" height="166" alt="image" src="https://github.com/user-attachments/assets/1ead9df9-00a0-4665-8ce3-af85af904afc" />

could you fix the rest of the conflicts

Dyslex7c commented:
@KolbyML hey I tried rebasing but it seems to bring back the obsolete code with the `spec_utils` and all and I tried fixing it but couldn't, need some help

KolbyML commented:
> @KolbyML hey I tried rebasing but it seems to bring back the obsolete code with the `spec_utils` and all and I tried fixing it but couldn't, need some help

I will give it a shot then
